Rename Shared Blocks to Reusable Blocks

noisysocks opened the pull request:
This is the first half of https://github.com/WordPress/gutenberg/issues/7887.

Renames all 'shared block' references to 'reusable block', and changes the copy of the associated UI controls to match what was suggested in https://github.com/WordPress/gutenberg/issues/7887#issuecomment-407390960.

To test:
1. Ensure tests pass
2. Click around in Gutenberg and look for any regressions

karmatosed commented:
Can we go with Reusable? Ideally saying 'Save this block to reuse' as the call? Apologies but I think this would work better going forward @noisysocks.

noisysocks committed:
Rename Shared Blocks to Reusable Blocks

noisysocks committed:
Add deprecated aliases for public reusable block functions

noisysocks commented:
I've updated this to use 'reusable', changed the copy of the UI controls to match what Kjell suggested in https://github.com/WordPress/gutenberg/issues/7887#issuecomment-407390960, and added deprecated aliases for the public selectors, actions and block API.

noisysocks committed:
Change copy surrounding reusable blocks

Change reusable block button copy so that 'reusable blocks' are treated
as a collection that blocks are added to or removed from.

youknowriad commented on editor/components/block-settings-menu/reusable-block-delete-button.js:
```diff
@@ -9,54 +9,54 @@ import { noop } from 'lodash';
 			role={ itemsRole }
 		>
-			{ __( 'Delete Shared Block' ) }
+			{ __( 'Remove from Reusable Blocks' ) }
```
This one is a bit confusing to me. I understand better "Remove reusable block" but I can see how the ambiguity there too.

youknowriad reviewed (approved):
LGTM 👍

noisysocks commented:
Gonna merge this to get ahead of merge conflicts 🙂 – can change the copy later if @karmatosed isn't OK with it.

aduth commented:
Any deprecations should be added to `docs/reference/deprecated.md`, which generates this page:

https://wordpress.org/gutenberg/handbook/reference/deprecated/

The functions deprecated here do not appear there and thus plugin authors who reference this page would not have the advance notice to have updated their usage.

aduth commented on editor/store/actions.js:
```diff
@@ -766,3 +767,63 @@ export function unregisterToken( name ) {
+	return updateReusableBlockTitle( id, title );
+}
+
+export function convertBlockToSaved( clientId ) {
```
This function was called `convertBlockToShared`, not `convertBlockToSaved`.

noisysocks replied:
🤦‍♂️ good catch
